src/geojson/geojson.js

```diff
@@ -0,0 +1,494 @@
+                    for (let y=0;y<json.coordinates[i].length;y++) {
+                        let out = "";
+                        let clone_coords = coords.cloneNode(true);
+                        clone_coords = clone_coords.querySelector("map-coordinates");
+
+                        // Going over coordinates for the polygon
+                        for (let x=0;x<json.coordinates[i][y].length;x++) {
+                            out = out + json.coordinates[i][y][x][0] + " " + json.coordinates[i][y][x][1] + " ";
+                        }
+
+                        // Create map-coordinates element and append it to clone_polygon
+                        clone_coords.innerHTML = out;
+
+                        clone_polygon.appendChild(clone_coords);
+                    }
+                    m.appendChild(clone_polygon);
+                }
+                return m;
+            case "GEOMETRYCOLLECTION": // ---------------------------------------------------------------------------
+                let g = geometrycollection.cloneNode(true);
+                g = g.querySelector('map-geometrycollection');
+                //console.log(json.geometries);
+                for (let i=0;i<json.geometries.length;i++) {
+                    let fg = geojson2mapml(json.geometries[i], options, layer);
+                    g.appendChild(fg);
+                }
+                return g;
+        }
+    }
+    return layer.querySelector('layer-');
+}
+
+// Takes an array of length n to return an array of arrays with length 2, helper function
+//    for mapml2geojson
+// breakArray: arr(float) -> arr(arr(float, float))
+function breakArray(arr) {
+    let size = 2; 
+    let arrayOfArrays = [];
+    // removing anything other than numbers, ., - (used to remove <map-span> tags)
+    arr = arr.filter(x => !(/[^\d.-]/g.test(x))).filter(x => x);
+    for (let i=0; i<arr.length; i+=size) {
+        arrayOfArrays.push((arr.slice(i,i+size)).map(Number));
+    }
+    return arrayOfArrays;
+}
+
+// Takes an HTML Table to return geojson properties, helper function
+//    for mapml2geojson
+// table2properties: HTML Table -> geojson
+function table2properties(table) {
+    let json = {};
+    table.querySelectorAll('tr').forEach((tr) => {
+        let k = tr.querySelector('th').innerHTML;
+        let v = tr.querySelector('td').innerHTML;
+        json[k] = v;
+    });
+    return json;
+}
+
+// Converts a geometry element to geojson, helper function
+//    for mapml2geojson
+// geometry2geojson: (child of <map-geometry>), Proj4, Proj4, Bool -> geojson
+function geometry2geojson(el, source, dest, transform) {
+    let elem = el.nodeName;
+    let j = {};
+    let coord;
+
+    switch(elem.toUpperCase()) {
+        case "MAP-POINT":
+            j.type = "Point";
+            if (transform) {
+                let pointConv = proj4.transform(source, dest, ((el.querySelector('map-coordinates').innerHTML.split(/[<>\ ]/g)).map(Number)) );

```

Comment:
I am getting a "proj4 is not defined" exception on this line when called like so:

```JavaScript
    let geojson = mapml2geojson(mapml, function(propertiesElement){
      return {"properties": propertiesElement.innerText};
    }, true);
```

Reply:
Good catch, looks like mapml-viewer.js does not fully get loaded in before that code is run. putting it inside a windows.onload event works for me. Will need to look for a solution...

`window.onload = (event) => {
  ...
};`